server/storage/mvcc/watchable_store.go

```diff
@@ -482,6 +496,27 @@ func (s *watchableStore) progress(w *watcher) {
 	}
 }
 
+func (s *watchableStore) progress_all(force bool) {
+	s.mu.RLock()
```

Comment:
`progress_all()` updates `s.progressOnSync`, so it's better to use `s.mu.Lock()`?
Note that I made sure that this line isn't related to the duplicated events issue. I'm still trying to understand why it happens.